controllers/machinehealthcheck_controller_test.go

```diff
@@ -306,16 +307,19 @@ var _ = Describe("MachineHealthCheck", func() {
 				for i, m := range machines {
 					targetMachines[i] = m.Name
 				}
+				sort.Strings(targetMachines)
 				// Make sure the status matches.
-				Eventually(func() int32 {
+				Eventually(func() *clusterv1.MachineHealthCheckStatus {
 					err := testEnv.Get(ctx, util.ObjectKey(mhc), mhc)
 					if err != nil {
-						return -1
+						return nil
 					}
-					return mhc.Status.CurrentHealthy
-				}).Should(Equal(int32(2)))
-				Expect(mhc.Status.ExpectedMachines).To(Equal(int32(2)))
-				Expect(mhc.Status.Targets).To(ConsistOf(targetMachines))
+					return &mhc.Status
+				}).Should(Equal(&clusterv1.MachineHealthCheckStatus{
```

Comment:
> Should(Equal(&clusterv1.MachineHealthCheckStatus

This is going to fail when we https://github.com/kubernetes-sigs/cluster-api/pull/3226 will merge.
However the same PR will bring custom matchers for  `ExpectedMachines` and `CurrentHealthy`, so I'm +1 to get a custom matcher for `Targets` as well as suggested by @detiber

Reply:
Can we do the checks inside the expect call and return true or false for now?

Reply:
Added a custom matcher but if it's undesired, can switch to doing all the checks in the Eventually. Only issue I see with that is for test failures we'll get crappy messages like expected false to be true

Reply:
Looks good, thanks!